packages/portal-proto/src/components/Table/VerticalTable.tsx

```diff
@@ -237,57 +288,57 @@ function VerticalTable<TData>({
                         key={header.id}
                         colSpan={header.colSpan}
                       >
-                        {flexRender(
-                          header.column.columnDef.header,
-                          header.getContext(),
-                        )}
+                        {headerName}
                       </th>
                     );
                   } else {
                     return (
                       <th
                         key={header.id}
-                        className={`${commonHeaderClass} whitespace-nowrap ${
+                        scope="col"
+                        className={`
+                        ${commonHeaderClass} whitespace-nowrap
+                        ${
                           isColumnHighlighted
                             ? "hover:bg-nci-purple-lighter"
                             : "hover:bg-primary-lightest"
-                        }`}
-                        onClick={() => {
-                          header.column.toggleSorting();
-                        }}
-                        onKeyDown={createKeyboardAccessibleFunction(() => {
-                          header.column.toggleSorting();
-                        })}
+                        }
+                      `}
                         aria-sort={
                           isColumnSorted
                             ? isColumnSorted === "desc"
                               ? "descending"
                               : "ascending"
-                            : undefined
+                            : "none"
                         }
                         colSpan={header.colSpan}
+                        role="columnheader"
                       >
-                        <button className="flex items-center font-heading">
-                          {flexRender(
-                            header.column.columnDef.header,
-                            header.getContext(),
+                        <button
+                          className="flex items-center font-heading"
+                          onClick={() =>
+                            handleSorting(header, headerName, isColumnSorted)
+                          }
+                          onKeyDown={createKeyboardAccessibleFunction(() =>
+                            handleSorting(header, headerName, isColumnSorted),
                           )}
```

Comment:
```suggestion
```
Keyboard navigation is currently not working, the above code is not needed when using a button as the keyboard input of Enter is translated automatically to an onclick event